src/jsonapi.js

```diff
@@ -55,24 +45,30 @@ export const uploadFile = (file, {
 } = {}) => {
   console.warn('uploadFile has been deprecated and will no longer be supported by redux-json-api https://github.com/dixieio/redux-json-api/issues/2');
 
+  if (onSuccess !== noop || onError !== noop) {
+    console.warn('onSuccess/onError callbacks are deprecated. Please use returned promise: https://github.com/dixieio/redux-json-api/issues/17');
+  }
+
   return (dispatch, getState) => {
-    const accessToken = getState().api.endpoint.accessToken;
+    const axiosConfig = getState().api.endpoint.axiosConfig;
     const path = [companyId, fileableType, fileableId].filter(o => !!o).join('/');
-    const url = `${__API_HOST__}/upload/${path}?access_token=${accessToken}`;
+    const endpoint = `${__API_HOST__}/upload/${path}`;
 
     const data = new FormData;
     data.append('file', file);
 
     const options = {
+      ... axiosConfig,
       method: 'POST',
       body: data
     };
 
-    return fetch(url, options)
-      .then(res => {
+    return apiRequest(endpoint, options)
+      .then((res) => {
+        onSuccess(res.data);
```

Comment:
@egeriis - Iam quite sure we already deleted these Callbacks in favor for just returning the promises.